### ISSUE TITLE
work loses collection association when edited by someone who doesn't own the collection

### ISSUE THREAD

**Issue by hackmastera** [chemheritage/chf-sufia#291]

**Comment by hackmastera** [comment]
replicated on dev

**Comment by hackmastera** [comment]
replicated on sufia core

**Comment by hackmastera** [comment]
As a temporary fix I'm changing line 54 of `/opt/sufia-project/shared/bundle/ruby/2.3.0/bundler/gems/curation_concerns-b073550c40a1/app/actors/curation_concerns/actors/add_to_collection_actor.rb` to check for an empty array instead of just the presence of an array.

This means that the function will return without effect as long as the person editing the record did not select a collection.

As long as there are no objects in more than one collection (or more specifically, collections owned by more than one person) this should hold us over until a more sustainable fix is implemented.

**Comment by hackmastera** [comment]
changing line from
`return true unless new_collection_ids`
to
`return true if new_collection_ids.empty?`

**Comment by hackmastera** [comment]
okay here's my new logic

``
        def add_to_collections(new_collection_ids)
          return true unless new_collection_ids
          # remove from old collections
          (curation_concern.in_collection_ids - new_collection_ids).each do |old_id|
            # don't remove from collections owned by other people
            collection = ::Collection.find(old_id)
            if user.eql? ::User.find_by_email collection.depositor
              collection.members.delete(curation_concern)
              collection.save
            end
          end
```

TODO:
- look again in curation concerns
- write test
- submit fix to CC 1.6.x
- port / submit to CC 1.7.x
- port / submit to CC master (2.0)
